src/shared/Microsoft.Git.CredentialManager/Settings.cs

```diff
@@ -55,20 +90,109 @@ public Settings(IEnvironmentVariables environmentVariables, IGit git)
 
         public bool IsSecretTracingEnabled => _environment.GetBooleanyOrDefault(KnownEnvars.GcmTraceSecrets, false);
 
+        public bool IsMsalTracingEnabled => _environment.GetBooleanyOrDefault(Constants.EnvironmentVariables.GcmTraceMsAuth, false);
+
+        public bool IsCertificateVerificationEnabled =>
+            TryGetSetting(null, KnownGitCfg.Http.SectionName, KnownGitCfg.Http.SslVerify, out string value) && !value.ToBooleanyOrDefault(true);
+
+        public Uri ProxyConfiguration
+        {
+            get
+            {
+                bool TryGetUriSetting(string envarName, string section, string property, out Uri uri)
+                {
+                    IEnumerable<string> allValues = GetSettingValues(envarName, section, property);
+
+                    foreach (var value in allValues)
+                    {
+                        if (Uri.TryCreate(value, UriKind.Absolute, out uri))
+                        {
+                            return true;
+                        }
+                    }
+
+                    uri = null;
+                    return false;
+                }
+
+                /*
+                 * There are several different ways we support the configuration of a proxy.
+                 *
+                 * In order of preference:
+                 *
+                 *   1. GCM proxy Git configuration
+                 *        credential.httpsProxy
+                 *        credential.httpProxy
+                 *
+                 *   2. Standard Git configuration
+                 *        https.proxy
+                 *        http.proxy
+                 *
+                 *   3. cURL environment variables
+                 *        HTTPS_PROXY
+                 *        HTTP_PROXY
+                 *        ALL_PROXY
+                 *
+                 *   4. Legacy GCM proxy environment variable
+                 *        GCM_HTTP_PROXY
+                 *
+                 * If the remote URI is HTTPS we check the HTTPS variants first, and fallback to the
+                 * non-secure HTTP options if not found.
+                 *
+                 * For HTTP URIs we only check the HTTP variants.
+                 *
+                 */
+
+                bool isHttps = StringComparer.OrdinalIgnoreCase.Equals(Uri.UriSchemeHttps, RemoteUri?.Scheme);
+
+                Uri proxyConfig;
+
+                // 1. GCM proxy Git configuration
+                if (isHttps && TryGetUriSetting(null, GitCredCfg.SectionName, GitCredCfg.HttpsProxy, out proxyConfig) ||
+                    TryGetUriSetting(null, GitCredCfg.SectionName, GitCredCfg.HttpProxy,  out proxyConfig))
+                {
+                    return proxyConfig;
```

Comment:
Is it worth logging that we used the GitCredCfg.HttpsProxy, setting and what the proxy was?  Same question for other 3 code paths.

Reply:
We log and warn the user that a legacy setting is being used now (including a link to how to configure the supported ways).